app/pages/Accounts/AccountDetailsPage/AccountReport/AccountReport.tsx

```diff
@@ -56,30 +69,66 @@ export default function AccountReport({ location }: Props) {
     const filtersRef = useRef<TransactionFiltersRef>(null);
 
     function promptDecrypt(account: Account) {
-        return new Promise((resolve) => {
+        return new Promise<{
+            decrypted: boolean;
+            address: string;
+            credentialNumber: number;
+            prfKeySeed: string;
+        }>((resolve) => {
             setShowDecrypt({
                 show: true,
                 header: decryptMessage(account.name),
                 account,
-                onFinish: (decrypted) => {
+                onFinish: (
+                    decrypted: boolean,
+                    address: string,
+                    credentialNumber: number,
+                    prfKeySeed: string
+                ) => {
                     setShowDecrypt({ show: false });
-                    resolve(decrypted);
+                    resolve({
+                        decrypted,
+                        address,
+                        credentialNumber,
+                        prfKeySeed,
+                    });
                 },
             });
         });
     }
 
-    /** Constructs the account report(s).
-     * If there are multiple chosen accounts, the reports will be bundled into a zip file
+    /**
+     * Constructs one or more account reports. If there are multiple chosen accounts,
+     * then the reports will be bundled into a zip file.
      */
     const makeReport = useCallback(
         async (filters: TransactionFilter) => {
+            let prfKeyMap: Record<string, CredentialNumberPrfKey> = {};
             const accountsToReport: Account[] = [];
             for (const account of accounts) {
-                const hasEncrypted = await containsEncrypted(account, filters);
-                if (!hasEncrypted || (await promptDecrypt(account))) {
-                    accountsToReport.push(account);
+                const hasEncrypted = hasEncryptedBalance(account);
+                const includeEncryptedTransfers =
+                    filters.encryptedAmountTransfer ||
+                    filters.encryptedAmountTransferWithMemo;
+                if (hasEncrypted && includeEncryptedTransfers) {
+                    const {
+                        decrypted,
+                        address,
+                        credentialNumber,
+                        prfKeySeed,
+                    } = await promptDecrypt(account);
+                    if (decrypted) {
+                        const updatedPrfKeyMap = {
+                            ...prfKeyMap,
+                        };
+                        updatedPrfKeyMap[address] = {
+                            credentialNumber,
+                            prfKeySeed,
+                        };
+                        prfKeyMap = updatedPrfKeyMap;
+                    }
                 }
+                accountsToReport.push(account);
```

Comment:
If i don't decrypt, then i get an error that credentialNumber is undefined. (but only the first time)

Reply:
I don't get this anymore.